src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1347,5 +1317,326 @@ private void VerifyNotepadMainWindowTitle(Process process, string filename)
+                            _process.Kill();
+
+                            Assert.True(_process.WaitForExit(WaitInMS));
+                        }
+
+                        if (PlatformDetection.IsNotWindowsServerCore)
+                        {
+                            // remove the access
+                            SetAccessControl((_processImpersonationAccount ?? _processAccount).AccountName, _process.StartInfo.FileName, _workingDirectory, add: false);
+                        }
+
+                        _additionalCleanup?.Invoke(_processAccount?.AccountName, _processImpersonationAccount?.AccountName);
+                        _processAccount?.Dispose();
+                        _processImpersonationAccount?.Dispose();
+                    }
+
+                    _disposedValue = true;
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(disposing: true);
+                GC.SuppressFinalize(this);
+            }
+        }
+    }
+
+    public partial class TestFileShare : IDisposable
+    {
+        private readonly string _shareName;
+
+        private readonly string _path;
+
+        private bool _disposedValue;
+
+        public TestFileShare(string shareName, string path)
+        {
+            _shareName = shareName;
+            _path = path;
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            SHARE_INFO_502 shareInfo = default;
+            shareInfo.shi502_netname = _shareName;
+            shareInfo.shi502_path = _path;
+            shareInfo.shi502_remark = "folder created to test UNC file paths";
+            shareInfo.shi502_max_uses = -1;
+
+            int infoSize = Marshal.SizeOf(shareInfo);
+            IntPtr infoBuffer = Marshal.AllocCoTaskMem(infoSize);
+
+            try
+            {
+                Marshal.StructureToPtr(shareInfo, infoBuffer, false);
+
+                int shareResult = NetShareAdd(string.Empty, 502, infoBuffer, IntPtr.Zero);
+                switch (shareResult)
+                {
+                    case 0:
+                        break;
+                    case 2118:
+                        NetShareDel(string.Empty, _shareName, 0);
+                        shareResult = NetShareAdd(string.Empty, 502, infoBuffer, IntPtr.Zero);
+                        break;
+                    default:
+                        throw new Exception($"Failed to create a file share, NetShareAdd returned {shareResult}");
+                }
+
+                if (shareResult != 0 && shareResult != 2118) // is a failure that is not a NERR_DuplicateShare
+                {
+                    throw new Exception($"Failed to create a file share, NetShareAdd returned {shareResult}");
+                }
+            }
+            finally
+            {
+                Marshal.FreeCoTaskMem(infoBuffer);
+            }
+        }
+
+        [LibraryImport(Interop.Libraries.Netapi32)]
```

Comment:
If given test or a utility is Windows specific, we usually make the class `partial`, and move the Windows specific parts to `$fileName.Windows.cs` and modify the project file to build it only for Windows:

https://github.com/dotnet/runtime/blob/9df6ea21007319967975dc9985413bb6518287da/src/libraries/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests.csproj#L44-L47